pkg/loghttp/push/otlp.go

```diff
@@ -135,6 +146,10 @@ func otlpToLokiPushRequest(ctx context.Context, ld plog.Logs, userID string, ten
 			return true
 		})
 
+		if !hasServiceName && len(discoverServiceName) > 0 && !stats.IsAggregatedMetric {
```

Comment:
I created the the variable and reused where I could, not sure if the line #s in your links were off, but it I don't think this can replace the `hasServiceName` logic, I think we need both